spec/system/placements/schools/partner_providers/add_a_provider_spec.rb

```diff
@@ -0,0 +1,262 @@
+require "rails_helper"
+
+RSpec.describe "School user adds a provider to their list of providers",
+               service: :placements, type: :system do
+  include ActiveJob::TestHelper
+
+  around do |example|
+    perform_enqueued_jobs { example.run }
+  end
+
+  scenario "User adds a provider", :js do
+    given_provider_exists
+    and_i_am_signed_in
+
+    when_i_am_on_the_providers_index_page
+    and_i_click_on_add_provider
+    and_i_click_on_continue
+    then_i_see_a_validation_error_for_entering_a_provider_search_term
+
+    given_shelbyville_university_is_already_in_my_list_of_providers
+    and_i_enter_shelbyville_university
+    then_i_see_a_dropdown_item_for_shelbyville_university
+    and_i_click_on_the_shelbyville_university_dropdown_item
+    and_i_click_on_continue
+    then_i_see_a_validation_error_for_shelbyville_university_is_already_in_my_list_of_providers
+
+    given_ogdenville_university_has_not_been_onboarded_to_the_placements_service
+    and_i_enter_ogdenville_university
+    then_i_see_a_dropdown_item_for_ogdenville_university
+    and_i_click_on_the_ogdenville_university_dropdown_item
+    and_i_click_on_continue
+    then_i_see_confirm_provider_details_page_for_ogdenville_university
+    and_i_click_on_confirm_and_add_provider
+    then_i_return_to_the_provider_index
```

Comment:
Rather than using this `then_i_return_to_the_provider_index` it might be worth following @Kizr's example, and do something like `then_i_see_the_providers_index_page`.